### DISCUSSION TITLE
Trouble Reading Firmware

### DISCUSSION THREAD

**Discussion by cpw7922** [emporia-vue-local/esphome#83] [Q&A]
Any time I try to read the ESP32 flash I get the following. It seems to always stop at 25%

```
D:\esp>esptool.py --no-stub -b 115200 read_flash 0 0x800000 flash_contents.bin

esptool.py v3.3.2
Found 2 serial ports
Serial port COM4
Connecting......
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting.......
Detecting chip type... ESP32
Chip is ESP32-D0WD (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 34:94:xxxxxxxx
Enabling default SPI flash mode...
2097152 (25%)
A fatal error occurred: Failed to read flash block (result was 01090000: CRC or checksum was invalid)
```

**Comment by flaviut** [comment]
3 main things to troubleshoot:

- [ ] are the solder joints for the pin header good?
- [ ] are the cables good?
- [ ] does using a different usb/serial adapter work?

**Reply by cpw7922** [reply]
are the solder joints for the pin header good? - Yes I believe so, it very repeatibly happens at the 25% mark. 
are the cables good? - Ditto above
does using a different usb/serial adapter work? - I have another one on order that should her here Thursday so I'm hoping that solves it

**Reply by csbence** [reply]
@cpw7922 Did the new serial interface resolve the issue? I'm facing the same problem.

**Reply by csbence** [reply]
@flaviut I've checked multiple serial adapters and getting the same results as @cpw7922. Do you have any other suggestions for this issue?

**Reply by cpw7922** [reply]
I actually gave up trying to back up the firmware. When I went to set it up using the stock Emporia firmware it wouldn't connect to wifi so Emporia sent me a new one. With a backup on the way I decided to just try and flash ESPHome and it ended up working. Not sure what the issue is but I did end up using the Espressif flasher. Still no idea why I couldn't back up but I was able to sucessfully flash ESPHome and it has been working great

https://www.espressif.com/en/support/download/other-tools

**Reply by flaviut** [reply]
No idea, could it be related to this issue? https://github.com/emporia-vue-local/esphome/discussions/31

**Reply by cpw7922** [reply]
> No idea, could it be related to this issue? #31

I don't think so because I was still able to flash just fine. Just couldn't backup the stock firmware for some reason. Maybe copy protection or something? Not sure if that's a thing on ESP chips
